lib/rules/no-attrs-in-components.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const utils = require('../utils/utils');
+
+const MESSAGE = 'Do not use this.attrs';
+
+//------------------------------------------------------------------------------
+// General rule - Don't use this.attrs
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Disallow usage of this.attrs in components',
+      category: 'General',
+      recommended: true
+    },
+    fixable: null,
+  },
+
+  create(context) {
+    return {
+      MemberExpression(node) {
```

Comment:
If this is not something that we already check, then I'd suggest postponing it. If it is, I'll add the check :)